### DISCUSSION TITLE
Inner border radius.

### DISCUSSION THREAD

**Discussion by mimoguz** [mkpaz/atlantafx#22] [Ideas]
Hi.

First, please let me say that you're doing a tremendous job. Kudos, and thank you!

Now, my suggestion: define an inner border radius config. Right now for many components' borders are drawn using multiple backgrounds. The problem is, both outer and inner backgrounds have the same radius. This causes noticeable artifacts with rounded borders, like this (at %400):

![checkbox1](https://user-images.githubusercontent.com/7283047/198110715-520fb4db-d7b3-42b7-adca-b52ba30135c9.png)

As you can see, borders are much thicker at corners.

I think this can safely be fixed by defining an inner corner radius like this (pardon my Sass):

```Sass
// _config.scss
$inner-border-radius: math.max($border-radius - $border-width, 0) !default;

// _checkbox.scss
-fx-background-radius: cfg.$border-radius, cfg.$inner-border-radius;
```

![checkbox2](https://user-images.githubusercontent.com/7283047/198111850-b17048a0-a1be-46a9-aadd-78a87ee2a444.png)

I have a fork with fixes if you want to take a look:  [https://github.com/mimoguz/atlantafx](https://github.com/mimoguz/atlantafx)

**Comment by mkpaz** [comment]
Thanks! This a really good suggestion. I noticed that border corners don't look good enough, but not tackled the problem yet. Right now I'm a little busy with another JavaFX project. Feel free to commit if you're willing to spare some time for this, I'll gladly accept the PR.

**Comment by mimoguz** [comment]
Sure 👍

**Comment by mimoguz** [comment]
Created a pull request. 

While testing my changes, I saw another issue with split spinners: 

![spinner](https://user-images.githubusercontent.com/7283047/198213468-cd0739aa-24bf-44e3-9953-9bb7eb1a89c6.png)

Is that -1 background inset intentional?

**Comment by mkpaz** [comment]
Nice catch! Fixed it. Probably it was, but became a bug after refactoring.
